openwisp-monitoring/files/monitoring.agent

```diff
@@ -276,11 +278,29 @@ main() {
 	INTERVAL=${INTERVAL:-300}
 	REGISTRATION_INTERVAL=$((INTERVAL / 10))
 	VERBOSE_MODE=${VERBOSE_MODE:-0}
+	BOOTUP_DELAY=${BOOTUP_DELAY:-10}
 	TMP_DIR="/tmp/openwisp/monitoring"
+	BOOTUP="$TMP_DIR/bootup"
 
 	[ -z "$MODE" ] && echoerr "missing required --mode option"
 
+	# make directory
+	mkdir -p "$TMP_DIR"
+
 	if [ "$MODE" = "collect" ]; then
+		if [ ! -f "$BOOTUP" ]; then
+			# actions to do only after agent start, not after restart
+			if [ "$BOOTUP_DELAY" -ne "0" ]; then
+				# get a random number between zero and $BOOTUP_DELAY
+				DELAY=$(/usr/sbin/openwisp-get-random-number 0 "$BOOTUP_DELAY")
+				logger "Delaying sending metrics for $DELAY seconds..." \
```

Comment:
Delaying sending metrics?
I would make the message more generic, eg: "Delaying initialization of the monitoring agent for $DELAY seconds."